AnEvilPenguin.Calculator/CalculatorLibrary/CalculatorLibrary.cs

```diff
@@ -0,0 +1,126 @@
+﻿using Newtonsoft.Json;
+using System.Diagnostics;
+
+namespace CalculatorLibrary
+{
+    public class Calculator
+    {
+        private List<Double> pastResults = new List<Double>();
+        private JsonWriter writer;
+
+        public Calculator()
+        {
+            StreamWriter logFile = File.CreateText("calculatorlog.json");
+            logFile.AutoFlush = true;
+            writer = new JsonTextWriter(logFile);
+            writer.Formatting = Formatting.Indented;
+            writer.WriteStartObject();
+            writer.WritePropertyName("Operations");
+            writer.WriteStartArray();
+        }
+
+        public void ClearHistory()
+        {
+            writer.WriteStartObject();
+            writer.WritePropertyName("ClearedHistory");
+            writer.WriteStartArray();
+
+            pastResults.ForEach(result => { writer.WriteValue(result); });
+
+            writer.WriteEndArray();
+            writer.WriteEndObject();
+
+            pastResults.Clear();
+        }
+
+        public double DoOperation(double num1, double num2, string op)
+        {
+            double result = double.NaN; // Default value is "not-a-number" if an operation, such as division, could result in an error.
```

Comment:
It's totally fine to make comments when you're studying, but whenever pushing code make sure your class members ( methods, fields, properties) and properties are named in a way that makes your code self-explanatory. Only make comments when absolutely necessary and there's something that can't be inferred from code.  Please review all comments and only leave the ones that are absolutely necessary for other devs (or your future self) :-)

Reply:
Makes sense!